### PR TITLE
[RISCV] Support multiple levels of truncates in combineTruncToVnclip.

### PR DESCRIPTION
**Pull request by topperc**
We can use multiple vnclips to saturate an i32 value into an i8 value.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -16263,22 +16263,39 @@ static SDValue combineTruncToVnclip(SDNode *N, SelectionDAG &DAG,
...
-      ClipOpc, DL, VT,
-      {Val, DAG.getConstant(0, DL, VT), DAG.getUNDEF(VT), Mask,
-       DAG.getTargetConstant(RISCVVXRndMode::RNU, DL, Subtarget.getXLenVT()),
-       VL});
+
+  MVT ValVT = Val.getSimpleValueType();
+
+  do {
+    MVT ValEltVT = MVT::getIntegerVT(ValVT.getScalarSizeInBits() / 2);
+    ValVT = MVT::getVectorVT(ValEltVT, ValVT.getVectorElementCount());
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVISelLowering.cpp:16289]: changeVectorElementType? 
Or we can add a helper `getHalfVectorElementSizeVT` like `getHalfNumVectorElementsVT`?

**Review by lukel97 — APPROVED** [review]

**Review by sun-jacobi — APPROVED** [review]
LGTM

**Review by preames — APPROVED** [review]
LGTM

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**